src/daemon/rpmostree-sysroot-upgrader.cxx

```diff
@@ -969,6 +980,13 @@ prep_local_assembly (RpmOstreeSysrootUpgrader *self,
     }
   else
     {
+      /* We still want to prepare() even if there's only enabled modules to validate.
```

Comment:
One thing I notice here is that the changes in this file `rpmostree-sysroot-upgrader.cxx` got *way* simpler right?  

And I think a good next step is to push down into `rpmostree-core.cxx` the concept of "only fetch repodata if there are packages/modules in the treefile" in which case I think even more of the complexity here drops away.